ethexe/service/src/lib.rs

```diff
@@ -514,6 +517,12 @@ impl Service {
                 Event::Consensus(event) => match event {
                     ConsensusEvent::ComputeAnnounce(announce) => compute.compute_announce(announce),
                     ConsensusEvent::PublishMessage(message) => {
+                        if let Some(rpc) = rpc.as_mut()
+                            && let SignedValidatorMessage::Promise(promise) = &message
+                        {
+                            rpc.receive_promise(promise.clone().into_data().payload);
```

Comment:
rpc.provide_promise / prc.resolve_promise or smth like this may be more suitable ?